### PR TITLE
[#IC-152] Updated repository to fp-ts io-ts 2

### PR DESCRIPTION
**Pull request by michaeldisaro**
#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: GetOnlineMerchants/optional_product_category_list_middleware.ts [hunk 1]
```diff
@@ -56,30 +58,28 @@ const CommaSeparatedListOf = (
...
-    )
-    .chain(maybeQuery =>
-      maybeQuery.foldL(
-        () => taskEither.of(none),
-        query =>
-          fromEither(
-            CommaSeparatedListOf(productCategoryCodec).decode(query)
-          ).bimap(ResponseErrorFromValidationErrors(ProductCategory), some)
+  pipe(
+    TE.of<IResponseErrorValidation, O.Option<unknown>>(
```
Comment (AleDore) [inline GetOnlineMerchants/optional_product_category_list_middleware.ts:70]: Do we need this?

File: middlewares/optional_header_param.ts [hunk 2]
```diff
@@ -20,21 +21,23 @@ import { fromEither, taskEither } from "fp-ts/lib/TaskEither";
...
-        () => taskEither.of(none),
-        header =>
-          fromEither(type.decode(header)).bimap(
-            ResponseErrorFromValidationErrors(type),
-            some
+) => (
+  request: Request
+): Promise<E.Either<IResponseErrorValidation, O.Option<A>>> =>
+  pipe(
+    TE.of<IResponseErrorValidation, O.Option<unknown>>(
```
Comment (AleDore) [inline middlewares/optional_header_param.ts:28]: ditto

File: middlewares/optional_query_param.ts [hunk 3]
```diff
@@ -20,21 +21,23 @@ import { fromEither, taskEither } from "fp-ts/lib/TaskEither";
...
-        () => taskEither.of(none),
-        query =>
-          fromEither(type.decode(query)).bimap(
-            ResponseErrorFromValidationErrors(type),
-            some
+) => (
+  request: Request
+): Promise<E.Either<IResponseErrorValidation, O.Option<A>>> =>
+  pipe(
+    TE.of<IResponseErrorValidation, O.Option<unknown>>(
```
Comment (AleDore) [inline middlewares/optional_query_param.ts:28]: ditto

Comment (michaeldisaro) [reply middlewares/optional_query_param.ts]: fixed in https://github.com/pagopa/io-functions-cgn-operator-search/pull/28/commits/8386c96583ac277673207742ac223f1fe84047fc

File: middlewares/optional_header_param.ts (hunk 2 above)
Comment (michaeldisaro) [reply middlewares/optional_header_param.ts]: fixed in https://github.com/pagopa/io-functions-cgn-operator-search/pull/28/commits/8386c96583ac277673207742ac223f1fe84047fc

File: GetOnlineMerchants/optional_product_category_list_middleware.ts (hunk 1 above)
Comment (michaeldisaro) [reply GetOnlineMerchants/optional_product_category_list_middleware.ts]: fixed in https://github.com/pagopa/io-functions-cgn-operator-search/pull/28/commits/8386c96583ac277673207742ac223f1fe84047fc

**Review by AleDore — APPROVED** [review]
LGTM

**Review by AleDore — APPROVED** [review]

**Pull request merged**